common/stats/stats.go

```diff
@@ -0,0 +1,479 @@
+// This package borrows from go-metrics so we can avoid exposing dependencies,
+// and to give callers more flexibility in recording and formatting scoped stats.
+//
+// Original license: github.com/rcrowley/go-metrics/blob/master/LICENSE
+//
+package stats
+
+import (
+	"encoding/json"
+	"log"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/rcrowley/go-metrics"
+)
+
+// For testing.
+var Time StatsTime = DefaultStatsTime()
+
+// Stats users can either reference this global receiver or construct their own.
+var CurrentStatsReceiver StatsReceiver = NilStatsReceiver()
+
+// Overridable instrument creation.
+var NewCounter func() Counter = newMetricCounter
+var NewGauge func() Gauge = newMetricGauge
+var NewGaugeFloat func() GaugeFloat = newMetricGaugeFloat
```

Comment:
I only did both since go-metrics provides both. In terms of usefulness, I
guess it mostly signals intent (I don't think we care about the
magnitude/precision tradeoff and we're not doing math with it). I'd keep it
by default but can remove it if Dan agrees.

On Mon, Jul 11, 2016 at 12:37 PM, Caitie McCaffrey <notifications@github.com
> wrote:

> In common/stats/stats.go
> <https://github.com/scootdev/scoot/pull/53#discussion_r70322154>:
>
> > +
> > +	"golang.org/x/net/context"
> > +
> > +	"github.com/rcrowley/go-metrics"
> > +)
> > +
> > +// For testing.
> > +var Time StatsTime = DefaultStatsTime()
> > +
> > +// Stats users can either reference this global receiver or construct their own.
> > +var CurrentStatsReceiver StatsReceiver = NilStatsReceiver()
> > +
> > +// Overridable instrument creation.
> > +var NewCounter func() Counter = newMetricCounter
> > +var NewGauge func() Gauge = newMetricGauge
> > +var NewGaugeFloat func() GaugeFloat = newMetricGaugeFloat
>
> Do we need both?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scootdev/scoot/pull/53/files/0455468671e9ac9d8a8c02d478eb32039d85d05a#r70322154>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABc5CUy7VZqPQmS28wP5Cw4jzJgCWc6_ks5qUpuQgaJpZM4JJmGF>
> .
>